sympy/utilities/iterables.py

```diff
@@ -1112,7 +1112,7 @@ def connected_components(G):
     References
     ==========
 
-    .. [1] https://en.wikipedia.org/wiki/Component_(graph_theory)
+    .. [1] https://en.wikipedia.org/wiki/Component_%28graph_theory%29
```

Comment:
Is these URL escapes necessary?

Reply:
Not necessary but I don't know why sphinx was reporting it as a broken link

Reply:
Personally I think it is a bug in sphinx build and not on our side and an issue should be posted on https://github.com/sphinx-doc/sphinx. I am reverting all the changes related to URL escapes.

Reply:
Does this work correctly on the rendered site? If not, then we need to make the change. Personally I don't think this is that big of a deal and it's probably best to escape here since parentheses are not always parsed as correctly as being part of the URL. This is a special instance where it could in principle know that `)` is part of the URL because of the syntax, but in general, if a URL is inline in the text, it would assume a `)` at the end of the URL is part of the text.

Reply:
![image](https://user-images.githubusercontent.com/77196231/219855991-dfbddd8d-20c5-482e-afab-ed42350569da.png)
@sylee957 @asmeurer I didn't realised it but it was an error in the current SymPy docs also. The URL is not rendered properly in the docs and what Aaron says seems to play out. In this case we need to encode the character `)` in the URL. I think it becomes necessary.